### PR TITLE
updateExperimentNameOrDescription using jdbi

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

**Review by conradoverta — APPROVED** [review]

**Review by liam-verta — APPROVED** [review]

File: backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java [hunk 1]
```diff
@@ -381,4 +391,96 @@ public FutureExperimentDAO(
...
+            projectIdFromExperimentMap ->
+                futureProjectDAO.checkProjectPermission(
+                    projectIdFromExperimentMap.get(request.getId()), ModelDBServiceActions.UPDATE),
+            executor)
+        .thenCompose(
+            unused -> {
+              var name = request.getName();
+              if (request.getName().isEmpty()) {
+                name = MetadataServiceImpl.createRandomName();
+              }
```
Comment (liam-verta) [inline backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java]: @conradoverta 
 When updating a name do we want to pick a random name if a blank one is passed?   I can see this behavior when creating a new experiment, but it seems an odd behavior when updating.

File: backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java [hunk 2]
```diff
@@ -381,4 +391,96 @@ public FutureExperimentDAO(
...
+            executor)
+        .thenCompose(unused -> getExperimentById(request.getId()), executor);
+  }
+
+  private InternalFuture<Void> updateExperimentName(String expId, String name) {
+    return jdbi.useHandle(
+        handle ->
+            handle
+                .createUpdate(
+                    "update experiment set name = :name, date_updated = :updatedTime, version_number=(version_number + 1) where id = :id ")
```
Comment (liam-verta) [inline backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java]: Similar to @conradoverta 's question one of the other PRs.  Does `name` need to be unique for all non-deleted experiments?

File: backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java [hunk 3]
```diff
@@ -381,4 +391,96 @@ public FutureExperimentDAO(
...
+    return getProjectIdByExperimentId(Collections.singletonList(request.getId()))
+        .thenCompose(
+            projectIdFromExperimentMap ->
+                futureProjectDAO.checkProjectPermission(
+                    projectIdFromExperimentMap.get(request.getId()), ModelDBServiceActions.UPDATE),
+            executor)
+        .thenCompose(
+            unused -> {
+              var name = request.getName();
+              if (request.getName().isEmpty()) {
```
Comment (liam-verta) [inline backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java]:
```suggestion
              var name = request.getName();
              if (name.isEmpty()) {
```

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java]: Added unique experiment name check in entire project.

**Pull request merged**